File: lintcode/best-time-to-buy-and-sell-stock-ii.cpp
#include <iostream>
#include <vector>
using namespace std;


class Solution {
		public:
		    /**
			 *      * @param prices: Given an integer array
			 *           * @return: Maximum profit
			 *                */

				int cal(vector<int> &prices,int n){
							int A = 0;
							if(n<=0) return 0;
							else A = prices[n] - prices[n-1];
							return A>0?(cal(prices,n-1)+A):cal(prices,n-1);
						}


			    int maxProfit(vector<int> &prices) {
							int n = prices.size();
							return cal(prices,n-1);
				    }
};

int main(){
	Solution sol;
	vector<int> v ;
	v.push_back(3);
	v.push_back(2);
	v.push_back(6);
	v.push_back(5);
	v.push_back(0);
	v.push_back(3);
	cout << sol.maxProfit(v) << endl;
	return 0;
}

File: pypractise/leetcode/best-time-to-buy-and-sell-stock-ii.py
__author__ = 'Daniel.Lee'
class Solution:
    # @param prices, a list of integer
    # @return an integer
    def maxProfit(self, prices):
        flag = False
        profit = 0
        if len(prices) < 2:
            return 0
        if prices[0] <= prices[1]:
            flag = True
        for i in range(1,len(prices)):
            if (not flag) and (prices[i] >= prices[i-1]):
                profit += (prices[i] - prices[i-1])
                flag = True
            elif flag:
                if prices[i] < prices[i-1]:
                    flag = False
                else:
                    profit += (prices[i] - prices[i-1])
        return profit

